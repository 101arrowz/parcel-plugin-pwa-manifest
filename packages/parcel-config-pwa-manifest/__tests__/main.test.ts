/// <reference types="./parcel" />
import Parcel from '@parcel/core';
import defaultConfig from '../index.json'
import { join } from 'path';
import { readFileSync, readdirSync } from 'fs';

process.chdir(__dirname);

const fp = (...p: string[]): string => join(__dirname, ...p);

const outDir = fp('tmp', Date.now().toString(36));
const cacheDir = join(outDir, 'cache');

const parcel = new Parcel({
  entries: [fp('index.html')],
  defaultConfig: {
    ...defaultConfig,
    extends: ['@parcel/config-default'],
    filePath: require.resolve('@parcel/config-default')
  },
  cacheDir,
  distDir: outDir
});
jest.setTimeout(60000);
test('Integrated correctly', async () => {
  await parcel.run();
  expect(readdirSync(outDir)).toMatchSnapshot();
  expect(readFileSync(join(outDir, 'index.html')).toString()).toMatchSnapshot();
  expect(
    readFileSync(join(outDir, 'browserconfig.xml')).toString()
  ).toMatchSnapshot();
  expect(
    JSON.parse(readFileSync(join(outDir, 'manifest.webmanifest')).toString())
  ).toMatchSnapshot();
});
