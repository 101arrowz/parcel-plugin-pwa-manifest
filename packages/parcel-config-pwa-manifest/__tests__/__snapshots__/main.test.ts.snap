// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrated correctly 1`] = `
Array [
  "__tests__.18ea1367.js",
  "__tests__.18ea1367.js.map",
  "apple-touch-icon.67e6641c.png",
  "browserconfig.xml",
  "cache",
  "icon-192x192.1421700b.png",
  "icon-192x192.60385309.webp",
  "icon-512x512.03718fd9.png",
  "icon-512x512.53c5ad04.webp",
  "index.html",
  "manifest.webmanifest",
  "mstile-150x150.db75c4ea.png",
  "mstile-310x150.df0ada1c.png",
  "mstile-310x310.7d35cb45.png",
  "mstile-70x70.d4870be2.png",
]
`;

exports[`Integrated correctly 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head><meta name=\\"msapplication-config\\" content=\\"./browserconfig.xml\\">,<meta name=\\"theme-color\\" content=\\"white\\">,<link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"./apple-touch-icon.67e6641c.png\\">,<link rel=\\"manifest\\" href=\\"./manifest.webmanifest\\">
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>My PWA</title>
</head>
<body>
  <script src=\\"/__tests__.18ea1367.js\\"></script>
</body>
</html>"
`;

exports[`Integrated correctly 3`] = `"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?><browserconfig><msapplication><tile><TileColor>white</TileColor><square70x70logo src=\\"./mstile-70x70.d4870be2.png\\"/><square150x150logo src=\\"./mstile-150x150.db75c4ea.png\\"/><square310x310logo src=\\"./mstile-310x310.7d35cb45.png\\"/><wide310x150logo src=\\"./mstile-310x150.df0ada1c.png\\"/></tile></msapplication></browserconfig>"`;

exports[`Integrated correctly 4`] = `
Object {
  "background_color": "white",
  "description": "A PWA",
  "display": "standalone",
  "icons": Array [
    Object {
      "sizes": "192x192",
      "src": "./icon-192x192.60385309.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "192x192",
      "src": "./icon-192x192.1421700b.png",
      "type": "image/png",
    },
    Object {
      "sizes": "512x512",
      "src": "./icon-512x512.53c5ad04.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "512x512",
      "src": "./icon-512x512.03718fd9.png",
      "type": "image/png",
    },
  ],
  "name": "my-pwa",
  "scope": "./",
  "short_name": "my-pwa",
  "start_url": "./",
  "theme_color": "white",
}
`;
